2-advanced-topics/data-pipelines/intro-data-pipelines.ipynb

```diff
@@ -0,0 +1,1044 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "a3ef6f62-8d24-4c15-a221-95d5be0a1dba",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "![Medallion Flowchart](assets/kenya_workflow.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "0b4c155b-f121-4555-b56d-1c5e36277dee",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "The above flow diagram of the jobs and dependencies between the jobs is obtained by adding the jobs and then adding the other jobs that the job at hand depends on.\n",
+    "\n",
+    "Note that for silver.py script to run, we need the bronze.py to run since we need to read the bronze table in silver.py. This shows the dependence of silver.py on bronze.py. Similarly, we can get all the dependencies as shown in the flow diagram above. This forms a directed acyclic graph (DAG) which ensures that no circular dependence is possible. \n",
+    "\n",
+    "The above diagram shows the successful (manual) run of the workflow -- the success of each job indicated by the green line at the top edge of each box. This also makes debugging easier since we can identify the failure of a job within this workflow.\n",
+    "\n",
+    "\n",
+    "This saves the user the difficulty of running these scripts **manually**. In addition to this, the workflow can be scheduled so that the entire collection runs in the correct order. This scheduling can be done either manually (i.e., run the workflow manually-- not the individual jobs), or set to run at specific intervals, or triggered when there is a new file added to a specific folder for instance, which makes the **automation** of this workflow extremely flexible. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "6c7e39b4-2a8f-401d-aa5d-007fb78c551e",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "### Lakehouse\n",
+    "\n",
+    "A lakehouse is a unified architecture which enables storage of structured and unstructured data in a unified system. The databricks lakehouse is a specific implementation, which offers tools to process this data in the lakehouse setup. \n",
+    "\n",
+    "It allows for unified data management, and more importantly avoids **data silos**. In our setting, the delta tables constructed are not necessarily tied to the project, and can be accessed across multiple projects. For instance, the table containing subnational population for Kenya can be accessed for the purposes of a different different project. "
+   ]
```

Comment:
Not sure if it's necessary to introduce lake house, especially towards the end. Consider remove, or move it up to where we first read from or write to the lake.